hs_core/views/resource_rest_api.py

```diff
@@ -0,0 +1,622 @@
+                    error_message = file_validation_dict.get('message', None)
+                    if not error_message:
+                        error_message = "Uploaded file(s) failed validation."
+
+        else:
+            error_msg = 'Resource creation failed.'
+            raise ValidationError(detail=error_msg)
+
+        # prepare response data
+        rest_api_response.status = 'CREATED'
+        rest_api_response.status_code = status.HTTP_201_CREATED
+        if error_message:
+            rest_api_response.message = 'Resource created. %s.' % error_message
+        else:
+            rest_api_response.message = 'Resource created.'
+
+        rest_api_response.data = {'resource_type': resource_type, 'resource_id': resource.short_id}
+        return Response(data=rest_api_response.dict,  status=status.HTTP_201_CREATED)
+
+
+class ScienceMetadataRetrieveUpdate(APIView):
+    """
+    Retrieve resource science metadata
+
+    REST URL: hsapi/scimeta/{pk}
+
+    HTTP method: GET
+
+    HTTP status code (on success): 200
+
+    :type pk: str
+
+    :param pk: id of the resource
+
+    :return: science metadata as xml string
+
+    :rtype: str
+
+    :raises:
+
+    &nbsp;&nbsp;&nbsp;NotFound: return json format: {'detail': 'No resource was found for resource id:pk'}
+
+    &nbsp;&nbsp;&nbsp;HTTP status code: 404
+
+    &nbsp;&nbsp;&nbsp;PermissionDenied: return json format: {'detail': 'You do not have permission to perform this action.'}
+
+    &nbsp;&nbsp;&nbsp;HTTP status code:403
+
+    REST URL: hsapi/scimeta/{pk}
+
+    HTTP method: PUT
+
+    HTTP status code (on success): 200
+
+    :return: resource id
+
+    :rtype: json of the format: {'resource_id':pk}
+
+    :type pk: str
+
+    :param pk: id of the resource
+
+    :raises:
+
+    &nbsp;&nbsp;&nbsp;NotFound: return json format: {'detail': 'No resource was found for resource id':pk}
+
+    &nbsp;&nbsp;&nbsp;HTTP status code: 404
+
+    &nbsp;&nbsp;&nbsp;PermissionDenied: return json format: {'detail': 'You do not have permission to perform this action.'}
+
+    &nbsp;&nbsp;&nbsp;HTTP status code:403
+
+    &nbsp;&nbsp;&nbsp;ValidationError: return json format: {parameter-1': ['error message-1'], 'parameter-2': ['error message-2'], .. }
+
+    &nbsp;&nbsp;&nbsp;HTTP status code:400
+    """
+
+    @property
+    def allowed_methods(self):
+        return ['GET', 'PUT']
+
+    def get(self, request, pk):
+        # check the user is authorized for this action and if the resource exists for the given resource id (pk)
+        utils.authorize(request, pk, view=True)
+
+        # TODO: once the science metadata xml file is available as a separate file on iRODS, that file needs to be returned
+        return Response(data=hydroshare.get_science_metadata(pk), status=status.HTTP_200_OK)
+
+    def put(self, request, pk):
+        # TODO: update science metadata - not sure what data will be passed to this api call
+        raise NotImplementedError()
```

Comment:
as stated above i think this whole view should be wrapped into the main resource GET.